plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaTable.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.deltalake.transactionlog.MetadataEntry;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getCheckConstraints;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnComments;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnInvariants;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnNames;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnTypes;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnsMetadata;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnsNullability;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getGeneratedColumnExpressions;
+import static java.util.Objects.requireNonNull;
+
+public class DeltaTable
+{
+    private final List<String> columnNames;
+    private final Map<String, Object> columnTypes;
+    private final Map<String, String> columnComments;
+    private final Map<String, Boolean> columnsNullability;
+    private final Map<String, Map<String, Object>> columnsMetadata;
+    private final Map<String, String> columnGenerations;
```

Comment:
wouldn't it be better to have a `List<...> columns` and then each column having name, type, comments, nullability, generation and metadata?